Unable to build Go package (scc) with -mod=readonly and vendored dependencies

In https://github.com/chimera-linux/cports/pull/768 @valpackett introduced `-mod=readonly -modcacherw` to `GOFLAGS`. It seems that `mod=readonly` might be undesirable if the project has vendored dependencies? The Go docs say:

> - -mod=readonly tells the go command to ignore the vendor directory and to report an error if go.mod needs to be updated.
> - -mod=vendor tells the go command to use the vendor directory. In this mode, the go command will not use the network or the module cache.
> By default, if the go version in go.mod is 1.14 or higher and a vendor directory is present, the go command acts as if -mod=vendor were used. Otherwise, the go command acts as if -mod=readonly were used.

— <https://go.dev/ref/mod>

I tried updating the `scc` package (which has a `vendor` directory) to a new version tonight and it failed to build with these new flags:

```
./cbuild -f pkg contrib/scc
=> scc-3.2.0-r0: running init_fetch hook: 000_env_pkg_config...
=> scc-3.2.0-r0: running pre_fetch hook: 000_update_check...
=> scc-3.2.0-r0: running do_fetch hook: 000_sources...
=> cbuild: updating software in /home/wmoore/Source/github.com/wezm/cports/bldroot container...
OK: 523 MiB in 53 packages
=> cbuild: removing autodeps...
=> scc-3.2.0-r0: building [go] for x86_64...
   [host] go: found (1.21.4-r0)
=> scc-3.2.0-r0: installing host dependencies: go=1.21.4-r0
=> scc-3.2.0-r0: running do_extract hook: 000_sources...
=> scc-3.2.0-r0: running do_prepare...
=> scc-3.2.0-r0: golang: vendor/ is present, skip download of modules
=> scc-3.2.0-r0: running do_patch hook: 000_patches...
=> scc-3.2.0-r0: running pre_configure hook: 000_script_wrapper...
=> scc-3.2.0-r0: running pre_configure hook: 001_prepare_users...
=> scc-3.2.0-r0: running do_build...
=> scc-3.2.0-r0: golang: go build -p 16 -v -trimpath -o /builddir/scc-3.2.0/build/
go: downloading github.com/spf13/cobra v1.6.1
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/dbaggerman/cuba v0.3.2
go: downloading github.com/minio/blake2b-simd v0.0.0-20160723061019-3f5f724cb5b1
go: downloading golang.org/x/text v0.8.0
go: downloading gopkg.in/yaml.v2 v2.4.0
processor/file.go:14:2: github.com/dbaggerman/cuba@v0.3.2: Get "https://proxy.golang.org/github.com/dbaggerman/cuba/@v/v0.3.2.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
processor/processor.go:17:2: github.com/json-iterator/go@v1.1.12: Get "https://proxy.golang.org/github.com/json-iterator/go/@v/v1.1.12.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
processor/workers.go:8:2: github.com/minio/blake2b-simd@v0.0.0-20160723061019-3f5f724cb5b1: Get "https://proxy.golang.org/github.com/minio/blake2b-simd/@v/v0.0.0-20160723061019-3f5f724cb5b1.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
processor/formatters.go:21:2: golang.org/x/text@v0.8.0: Get "https://proxy.golang.org/golang.org/x/text/@v/v0.8.0.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
processor/formatters.go:22:2: golang.org/x/text@v0.8.0: Get "https://proxy.golang.org/golang.org/x/text/@v/v0.8.0.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
processor/formatters.go:23:2: gopkg.in/yaml.v2@v2.4.0: Get "https://proxy.golang.org/gopkg.in/yaml.v2/@v/v2.4.0.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
main.go:11:2: github.com/spf13/cobra@v1.6.1: Get "https://proxy.golang.org/github.com/spf13/cobra/@v/v1.6.1.zip": dial tcp: lookup proxy.golang.org on 10.0.0.1:53: dial udp 10.0.0.1:53: connect: network is unreachable
```

If add the following to the template then it builds:

```python
tool_flags = {"GOFLAGS": ["-mod=vendor"]}
```

I'm not really familiar with Go so I'm not sure whether this is a cbuild issue or somthing the template should handle (as above).

--- a/contrib/scc/template.py
+++ b/contrib/scc/template.py
@@ -1,6 +1,6 @@
 pkgname = "scc"
-pkgver = "3.1.0"
-pkgrel = 1
+pkgver = "3.2.0"
+pkgrel = 0
 build_style = "go"
 hostmakedepends = ["go"]
 pkgdesc = "Fast and accurate code counter"
@@ -8,7 +8,7 @@ maintainer = "Wesley Moore <wes@wezm.net>"
 license = "MIT OR Unlicense"
 url = "https://github.com/boyter/scc"
 source = f"{url}/archive/v{pkgver}.tar.gz"
-sha256 = "bffea99c7f178bc48bfba3c64397d53a20a751dfc78221d347aabdce3422fd20"
+sha256 = "69cce0b57e66c736169bd07943cdbe70891bc2ff3ada1f4482acbd1335adbfad"
 # objcopy fails on ppc
 options = ["!debug"]
 

--- a/src/cbuild/core/profile.py
+++ b/src/cbuild/core/profile.py
@@ -245,7 +245,7 @@ def _get_rustflags(
 
 def _get_goflags(self, name, extra_flags, debug, hardening, opts, stage, shell):
     hard = _get_harden(self, hardening, opts, stage)
-    bflags = ["-mod=readonly", "-modcacherw"]
+    bflags = ["-modcacherw"]
 
     if hard["pie"]:
         bflags.append("-buildmode=pie")

